src/app/product-managment/product-list/products-table/products-table.component.ts

```diff
@@ -3,44 +3,35 @@ import { MyService } from '../../product.service';
 import { ProductModel } from '../../create-product/product.model';
 import { MatTableDataSource } from '@angular/material/table';
 
-
 @Component({
   selector: 'app-products-table',
   templateUrl: './products-table.component.html',
   styleUrls: ['./products-table.component.scss']
 })
 export class ProductsTableComponent implements OnInit, OnChanges {
   public displayedColumns: string[] = ['name', 'tags', 'categories', 'description', 'edit', 'delete'];
-  public products: MatTableDataSource<ProductModel[]> = new MatTableDataSource();
-  @Input() private listOfItems;
-  @Input() private filterDate: string;
+  public arrayOfItems: MatTableDataSource<ProductModel[]> = new MatTableDataSource();
   public constructor(private myService: MyService) {
   }
 
   public ngOnInit(): void {
     this.myService.getProducts().subscribe(list => {
-      this.products = list.items as unknown as MatTableDataSource<ProductModel[]>
+      this.arrayOfItems = list.items as unknown as MatTableDataSource<ProductModel[]>;
     })
   }
 
    public ngOnChanges(_changes: SimpleChanges): void{
     if(_changes.listOfItems.currentValue !== _changes.listOfItems.previousValue){
```

Comment:
you are comparing refereneces here not actual aray content. It should be changed